src/classes/BDI_ManageAdvancedMappingCtrl.cls

```diff
@@ -0,0 +1,402 @@
+    * Set name and converts each to instances of DataImportFieldMappingWrapper for use in the UI.
+    *
+    * @param objectName: Object name
+    * @param fieldMappingSetname: Field Mapping Set name
+    *
+    * @return List: List of DataImportFieldMappingWrapper
+    */
+    @AuraEnabled
+    public static DataImportFieldMappingWrapper[] getFieldMappingsByObjectAndFieldSetNames(
+        String objectName, String fieldMappingSetname) {
+
+        DataImportFieldMappingWrapper[] fieldMappingWrappers =
+            new List<DataImportFieldMappingWrapper>();
+
+        Data_Import_Field_Mapping__mdt[] fieldMappings =
+            bdiCMT.fieldMappingsByObjMappingDevName.get(objectName);
+
+        for (Data_Import_Field_Mapping__mdt fieldMapping : fieldMappings) {
+
+            if (fieldMapping.Data_Import_Field_Mapping_Set__r.DeveloperName == fieldMappingSetname &&
+                fieldMapping.Is_Deleted__c == false) {
+
+                fieldMappingWrappers.add(new DataImportFieldMappingWrapper(fieldMapping));
+            }
+        }
+
+        return fieldMappingWrappers;
+    }
+
+    /*******************************************************************************************************
+    * @description Converts a JSON string of a DataImportFieldMappingWrapper to an instance of
+    * Data_Import_Field_Mapping__mdt. Replaces the prefix 'xxx_' with the namespace if necessary
+    * and replaces the suffix '_xxx' with '__c' if necessary.
+    *
+    * @param fieldMappingString: DataImportFieldMappingWrapper as JSON
+    *
+    * @return String: Deployment Id
+    */
+    @AuraEnabled
+    public static String createDataImportFieldMapping(String fieldMappingString) {
+        fieldMappingString = fieldMappingString.replaceAll('_xxx','__c');
+
+        if (NAMESPACE == 'npsp') {
+            fieldMappingString = fieldMappingString.replaceAll('xxx_','npsp__');
+        } else {
+            fieldMappingString = fieldMappingString.replaceAll('xxx_','');
+        }
+
+        try {
+            Data_Import_Field_Mapping__mdt difm =
+                (Data_Import_Field_Mapping__mdt)JSON.deserialize(
+                    fieldMappingString, Data_Import_Field_Mapping__mdt.class);
+
+            String deploymentId = enqueueMetadataRecords(difm);
+
+            return deploymentId;
+        } catch (Exception e) {
+            throw e;
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description Method takes in a list of generic sObjects and casts them a relevant metadata type.
+    * Builds the deployment container, metadata type records along with their fields, and enqueues them
+    * for deployment.
+    *
+    * @param customMetadataList: List of generic sObjects to be converted to custom metadata type records
+    *
+    * @return String: Deployment Id
+    */
+    public static String enqueueMetadataRecords(List<sObject> customMetadataList ) {
```

Comment:
This is great that this is super-generic, which maybe would imply this should be more broadly accessible in a utility type location? Or does that start to get in the way of existing utility methods? We definitely can approach that shift for future refactor as well.

Reply:
Could we potentially put this method into the CMT_MetadataAPI class after cleaning it up? @ptewson-sfdo